### PR TITLE
feat: ebook record dedup

### PR DESCRIPTION
**Pull request by BadrAly**
How to test:
- git PR
- run_test.sh
- if this a new instance make sure to run bootstrap and setup
- run the following command to update existing records
dojson -l marcxml -i data/ebooks.xml do cantookmarc21 | invenio records create_else_update --vendor cantook

Signed-off-by: Aly Badr <aly.badr@rero.ch>
Co-authored-by: Gianni Pante <gianni.pante@rero.ch>

### CODE REVIEW

File: tests/api/test_api.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    assert cdf_status == 'created'
+    assert new_cdf_record['electronic_location_and_access'][0][
+        'uniform_resource_identifier'
+    ]
+    merged_record, merged_record_status = Ebook.create_or_update(
+        mv_record, vendor='cantook', dbcommit=True, reindex=True)
+    assert merged_record_status == 'updated'
+    ela = merged_record['electronic_location_and_access']
+    assert len(ela) == 2
+    first_856 = ela[0]['uniform_resource_identifier'][0]
```
Comment (jma) [inline tests/api/test_api.py]: do not use 856 here!

File: rero_ebooks/api.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        id_=None,
+        delete_pid=True,
+        dbcommit=False,
+        reindex=False,
+        vendor=None,
+        **kwargs
+    ):
+        """Create a new ebook record."""
+        assert cls.minter
+        if delete_pid and data.get('pid'):
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ebooks/api.py]: accepted your proposition, done

File: rero_ebooks/api.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            del (data['pid'])
+        if not id_:
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ebooks/api.py]: problem copy and paste from reroils_data, fixed

File: rero_ebooks/api.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        )
+        try:
+            persistent_identifier, record = resolver.resolve(str(pid))
+            return super(Ebook, cls).get_record(
+                persistent_identifier.object_uuid, with_deleted=with_deleted
+            )
+        except PIDDoesNotExistError:
+            return None
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ebooks/api.py]: accepted your proposition, done

File: tests/api/conftest.py [hunk 5]
```diff
@@ -24,3 +24,75 @@ def app_config(app_config):
...
+                    '5788be89dde6b2d458f42b35'
+                ],
+            }
+        ],
+    }
+
+
+@pytest.yield_fixture()
+def mv_record():
+    """CDF record."""
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply tests/api/conftest.py]: MV: mediatheque-valais, comments added to the tests

File: tests/api/conftest.py [hunk 6]
```diff
@@ -24,3 +24,75 @@ def app_config(app_config):
     """Overwrite default configuration."""
     app_config['PIDSTORE_RECID_FIELD'] = 'pid'
     return app_config
+
+
+@pytest.yield_fixture()
+def cdf_record():
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply tests/api/conftest.py]: cdf = la-chaux-de-fonds, comments added to the tests

File: tests/api/test_api.py (hunk 1 above)
Comment (BadrAly) [reply tests/api/test_api.py]: used uri instead of 856, done

**Pull request merged**